test/tests/carousel_carousel-2-tablist.js

```diff
@@ -0,0 +1,364 @@
+'use strict';
+
+const { ariaTest } = require('..');
+const { By, Key } = require('selenium-webdriver');
+const assertAttributeValues = require('../util/assertAttributeValues');
+const assertAriaControls = require('../util/assertAriaControls');
+const assertAriaLabelledby = require('../util/assertAriaLabelledby');
+const assertAriaLabelExists = require('../util/assertAriaLabelExists');
+const assertAriaRoles = require('../util/assertAriaRoles');
+const assertTabOrder = require('../util/assertTabOrder');
+
+const exampleFile = 'carousel/carousel-2-tablist.html';
+
+const ex = {
+  rotationSelector: '#ex1 .rotation',
+  landmarkSelector: '#myCarousel',
+  tablistSelector: '#ex1 [role="tablist"]',
+  tabSelector: '#ex1 [role="tab"]',
+  tabpanelSelector: '#ex1 [role="tabpanel"]',
+  tabCount: 6,
+  tabTabOrder: [
+    // tab id, tabpanel id
+    ['#carousel-tab-1', '#carousel-item-1'],
+    ['#carousel-tab-2', '#carousel-item-2'],
+    ['#carousel-tab-3', '#carousel-item-3'],
+    ['#carousel-tab-4', '#carousel-item-4'],
+    ['#carousel-tab-5', '#carousel-item-5'],
+    ['#carousel-tab-6', '#carousel-item-6']
+  ],
+  rotationLabelPlaying: 'Stop automatic slide show',
+  rotationLabelPaused: 'Start automatic slide show'
+};
+
+const openTabAtIndex = async function (t, index) {
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  await tabs[index].click();
+};
+
+const waitAndCheckFocus = async function (t, selector, index) {
+  return t.context.session.wait(async function () {
+    return t.context.session.executeScript(function () {
+      const [selector, index] = arguments;
+      let items = document.querySelectorAll(selector);
+      return items[index] === document.activeElement;
+    }, selector, index);
+  }, t.context.waitTime, 'Timeout waiting for document.activeElement to become item at index ' + index + ' of elements selected by: ' + selector);
+};
+
+const waitAndCheckAriaSelected = async function (t, index) {
+  return t.context.session.wait(async function () {
+    const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+    return (await tabs[index].getAttribute('aria-selected')) === 'true';
+  }, t.context.waitTime, 'Timeout waiting for aria-selected to be set to true.');
+};
+
+// Attributes
+
+ariaTest('Carousel 2: rotation button has aria-label that is updated based on pause state', exampleFile, 'rotation-aria-label', async (t) => {
+  t.plan(4);
```

Comment:
`t.plan()` isn't necessary in these tests, since they are effectively synchronous with `async`/`await`, and the docs seem to [discourage it](https://github.com/avajs/ava/blob/master/docs/recipes/when-to-use-plan.md)

Reply:
Good point, `t.plan()` is unnecessary in this case and many others -- but I'm not convinced it's bad to leave it in. It is reasonable to allow/encourage `t.plan()` everywhere because some tests need it, and it's an easy rule to express and follow (the rule: always use `t.plan()`?).

Reply:
Anecdotally, this was confusing to me when I first starting looking at the existing tests in this repo, because it was so ubiquitous yet didn't seem to have an obvious reason. It also has the drawback that when modifying tests, it acts as an unnecessary gotcha -- unless you remember to update `t.plan()` when changing the number of tests run, you get a failure.

I don't think it would be too confusing to those writing tests to simply leave it out unless it's truly needed. It looks like the vast majority of our tests do not require it, since we're not doing anything fancy with interdependent async callbacks.

Reply:
Right, it is annoying to update t.plan() when you change the number of assertions on purpose, but maybe not as annoying as false positives? The more common case we face in these tests are the for loops (and maybe some other branching) -- the test passes because the loop is over nothing due to do a change in the code that breaks the selector, for example.

Reply:
Since the primary motivation to keep `t.plan()` is to catch empty selectors that then silently fail when iterating over a loop, how about this as a solution?: we create a helper to get elements that also has an assertion that the returned elements array is not empty.

It would have two benefits: we don't need to worry about people forgetting to add `t.plan()`, and we also aren't testing the number of elements that exist in the DOM, as long as there's at least one. So, as an example, if we end up updating the tabs example to have one more tab, we don't then need to go update a bunch of `t.plan()`s in the tests.

What do you think? I'd be happy to do the work for this if you like the approach.

Reply:
I’m also fine with this not being addressed in this PR

Reply:
@smhigley this seems like a fine solution to me.